src/clj/kixi/hecuba/api/devices.clj

```diff
@@ -118,17 +118,45 @@
 
 (defmethod calculated-sensor :default [sensor])
 
+(defn have-sensor?
+  "Check if readings have a sensor w/ a particular period."
+  [sensors period]
+  (some #(= period (:period %)) sensors))
+
+(defn get-sensor
+  "Retrieve a sensor map of a particular period in a sequence of readings."
+  [sensors period]
+  (filter #(= period (:period %)) sensors))
+
+(defn create-synth-sensors-new
+  "Added to fix the absence of some synthetic sensors."
+  [sensors]
+  (let [cumulative-sensors (get-sensor sensors "CUMULATIVE")]
+    (let [pulse-sensors (map #(ext-type % "differenceSeries")
+                             cumulative-sensors)
+          calc-sensors (map #(calculated-sensor %) pulse-sensors)]
+      (vector cumulative-sensors pulse-sensors calc-sensors))))
+
+(defn create-synth-sensors-default
+  "The original way of creating synthetic sensors.
+  It called throughout the api."
+  [sensors]
+  (concat sensors (map #(case (:period %)
+                          "CUMULATIVE" (ext-type % "differenceSeries")
+                          "PULSE"      (calculated-sensor %)
+                          "INSTANT"    nil
+                          nil) sensors)))
```

Comment:
Why doesn't `create-synth-sensors-new` just replace `create-synth-sensors-default`? I'm not sure why we need 2 functions. Can we do something like:

```
(->> sensors
    difference-series-sensors
    calculated-sensors)
```

That way all the PULSE sensors would get created from any existing CUMULATIVE sensors in the :readings and then all the calculated sensors could be created from the back of that.

wdyt?